sdk/resourcemanager/Azure.ResourceManager.Core/src/GenericResourceData.cs

```diff
@@ -9,7 +9,7 @@ namespace Azure.ResourceManager.Core
     /// <summary>
     /// A class representing the generic azure resource data model.
     /// </summary>
-    public class GenericResourceData : TrackedResource<ResourceManager.Resources.Models.GenericResource>
+    public class GenericResourceData : TrackedResource<TenantResourceIdentifier, ResourceManager.Resources.Models.GenericResource>
```

Comment:
All concrete resource types derive from TenantResourceIdentifier, the only id that does not is RootResourceIdentifier, so this specifically excludes RootResourceIdentifier. otherwise, using ResourceIdentifier is not functionally different.